Sources/StreamChat/Workers/ChannelListCleanupUpdater.swift

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+class ChannelListCleanupUpdater<ExtraData: ExtraDataTypes>: Worker {
+    // MARK: - Properties
+    
+    private let channelUpdater: ChannelListUpdater<ExtraData>
+    
+    // MARK: - Init
+    
+    override init(
+        database: DatabaseContainer,
+        apiClient: APIClient
+    ) {
+        channelUpdater = ChannelListUpdater(database: database, apiClient: apiClient)
+        super.init(
+            database: database,
+            apiClient: apiClient
+        )
+    }
+    
+    init(
+        database: DatabaseContainer,
+        apiClient: APIClient,
+        channelListUpdater: ChannelListUpdater<ExtraData>
+    ) {
+        channelUpdater = channelListUpdater
+        super.init(
+            database: database,
+            apiClient: apiClient
+        )
+    }
+    
+    func cleanupChannelList() {
+        database.write { _ in
+            self.allChannels.forEach { $0.syncFailedCleanUp() }
+        } completion: { error in
+            if let error = error {
+                log.error("Failed cleaning up channels data: \(error).")
+            }
+            self.updateChannels()
```

Comment:
I thought it makes sense to hold self with strong reference here so we won't end up in some half-cleaned undefined state